File: Programmers/src/JS/최소직사각형.js
function solution(sizes) {
    let answer = 0;
    let tmp = 0;
    let max_x = 0, max_y=0;
    
    for(let i=0; i<sizes.length; i++){
        if(sizes[i][0]<sizes[i][1]){
            tmp = sizes[i][0];
            sizes[i][0]=sizes[i][1];
            sizes[i][1]=tmp;
        }
    }
    
    for(let i=0; i<sizes.length; i++){
        max_x = Math.max(max_x, sizes[i][0]);
        max_y = Math.max(max_y, sizes[i][1]);
    }
    
    answer = max_x * max_y;
    
    return answer;
}

File: Programmers/src/Java/Lv1/최소직사각형.java
class Solution {
    public int solution(int[][] sizes) {
        int answer = 0;
        int width = 0, height = 0;
        
        for(int i=0; i<sizes.length; i++) {
            if(sizes[i][0] < sizes[i][1]) {
                int tmp = sizes[i][0];
                sizes[i][0] = sizes[i][1];
                sizes[i][1] = tmp;
            }
            
            width = Math.max(sizes[i][0], width);
            height = Math.max(sizes[i][1], height);
        }
        
        answer = width * height;
        
        return answer;
    }
}